### PR TITLE
build: Replace `Yarn` with `PNPM`

### PR DESCRIPTION
**Pull request by morgsmccauley**
## Motivation
#972 

## Description
This PR migrates the repos package manager from `yarn` to `pnpm`. The migration was mostly seamless, but there were a few issues relating to missing/misplaced dependencies, I have commented in-line highlighting these.

Rather than migrating `cookbook` separately, I have moved it under `packages/` so it can be managed by `pnpm` workspaces. This means `cookbook` will be bootstrapped, as well as all other `packages`, when running `pnpm install`. It will also manage the symlinks between shared `packages`.

## Checklist
- [x] Read the [contributing](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md) guidelines
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [x] Performed a self-review of the PR
- [ ] Added automated tests
- [ ] Manually tested the change

### CODE REVIEW

File: packages/cookbook/package.json [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
   "author": "",
   "license": "ISC",
   "dependencies": {
     "chalk": "^4.1.1",
     "homedir": "^0.6.0",
-    "near-api-js": "file:../../"
+    "near-api-js": "workspace:*"
```
Comment (morgsmccauley) [inline packages/cookbook/package.json:12]: This indicates to `pnpm` that it should create a symlink to the `near-api-js` package within the workspace.

File: packages/cookbook/package.json [hunk 2]
```diff
@@ -1,16 +1,14 @@
 {
-  "name": "cookbook",
+  "name": "@near-js/cookbook",
+  "private": true,
```
Comment (morgsmccauley) [inline packages/cookbook/package.json:3]: Prevents this package from being published.

File: packages/near-api-js/package.json [hunk 3]
```diff
@@ -26,12 +26,12 @@
     "devDependencies": {
         "@types/bn.js": "^5.1.0",
         "@types/http-errors": "^1.6.1",
+        "@types/node": "^18.7.14",
```
Comment (morgsmccauley) [inline packages/near-api-js/package.json:29]: This was a [phantom dependency](https://rushjs.io/pages/advanced/phantom_deps/#phantom-dependencies) 🙀 lucky it was only types.

File: packages/near-api-js/package.json [hunk 4]
```diff
@@ -26,12 +26,12 @@
...
         "browserify": "^16.2.3",
         "bundlewatch": "^0.3.1",
         "concurrently": "^7.3.0",
         "danger": "^11.1.1",
         "danger-plugin-yarn": "^1.3.2",
-        "husky": "^7.0.4",
```
Comment (morgsmccauley) [inline packages/near-api-js/package.json:34]: Moved to workspace root.

File: packages/near-api-js/package.json [hunk 5]
```diff
@@ -40,28 +40,27 @@
         "semver": "^7.1.1",
         "ts-jest": "^26.5.6",
         "ts-morph": "^15.1.0",
-        "typescript": "^4.7.4",
```
Comment (morgsmccauley) [inline packages/near-api-js/package.json:43]: Moved to workspace root as it's used across the repo.

File: package.json [hunk 6]
```diff
@@ -2,16 +2,8 @@
   "name": "@near-api-js/monorepo",
   "private": true,
   "engines": {
-    "node": ">=16.14.0"
-  },
-  "workspaces": {
-    "packages": [
-      "packages/*"
-    ],
-    "nohoist": [
```
Comment (morgsmccauley) [inline package.json:11]: This hoisting is no longer a concern - pnpm doesn't hoist all package dependencies to the workspace root like yarn does, all packages have their own node_modules folder with all their defined dependencies (but these are all symlinked for perf reasons).

**Review by hcho112 — APPROVED** [review]
LGTM, had brief conversation with @morgsmccauley on benefit of using `pnpm` 

1. It stores packages in single root location. (Packages do have node_modules directory but its sim linked to root pnpm package directory
2. Stop from accidentally using phantom dependencies. (only use packages that are installed) 

Good work!

**Pull request merged**